### ISSUE TITLE
Helm chart version1.16.0-rc.0 breaks GatewayAPI hostNetwork

### ISSUE THREAD

**Issue by Silvest89** [cilium/cilium#33332]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

gatewayAPI.externalTrafficPolicy is added but this won't work when hostNetwork is enabled

cilium-configmap.yaml
```
gateway-api-service-externaltrafficpolicy: {{ .Values.gatewayAPI.externalTrafficPolicy | quote }}
```

Neither `Local` or `Cluster` will work but it is required since it has a validation added

```
{{- if .Values.gatewayAPI.enabled }}
  {{- if not (or (eq .Values.gatewayAPI.externalTrafficPolicy "Cluster") (eq .Values.gatewayAPI.externalTrafficPolicy "Local")) }}
    {{ fail "Cilium GatewayAPI needs an externalTrafficPolicy set to 'Cluster' or 'Local'." }}
  {{- end }}
{{- end }}
```
```
cilium gateway api unable to create service externaltrafficpolicy may only be set for externally-accessible service
```

It's added in 1.16.0-rc.0, chart version 1.16.0-pre.3 works fine.

### Cilium Version

1.16.0-rc.0

### Kernel Version

.

### Kubernetes Version

1.30.2

### Regression

1.16.0-pre.3

### Cilium Users Document

- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by youngnick** [comment]
#33101 seems like it will address this one. Thanks for logging the issue @Silvest89.

**Comment by otaconix** [comment]
#33101 has been merged, and is available under [v1.16.0-rc.1](https://github.com/cilium/cilium/releases/tag/v1.16.0-rc.1). I assume this fixes the issue. Let us know if it does for you!

**Comment by aanm** [comment]
Closing since it has been fixed with #33101
